### PR TITLE
PR: Add a file switcher instance

### LINKED ISSUE #45: Add a file switcher instance to this plugin
It'd be really nice to have it to switch among notebooks.

@goanpeca, question: `Ctrl+P` (the shortcut to call the switcher) is a global one right now, but I think we should use the same one here. So what do you think we should do to make that work?

Possibilities:

1. Make `Ctrl+P` a plugin shortcut. Cons: Not nice because you'd have to give focus to the Editor or the notebook to make it work.
2. Make `Ctrl+P` to detect the current plugin and change its behavior accordingly.
3. Add an API for the file switcher so external plugins can register with it.

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #45 

Preview:

fs
___________________________________________

In order for this to work there is a need to modify the FileSwitcher class in spyder to map the correct methods to get the current widget (client).

The methods to change will be:
* `get_editor` (change the call for the `get_current_editor`)
* `save_initial_state` (let the current widget not have `textCursor`)

Also this works for now by adding a button to open the fileswitcher.

### CODE REVIEW

**Comment by ccordoba12** [comment]
@dalthviz, please make the corresponding changes in the Spyder repo against the `3.1.x` branch.

**Comment by ccordoba12** [comment]
Very good job, by the way!

**Comment by goanpeca** [comment]
great job :-p, this was accomplished in fewer lines than I expected :-)

**Comment by ccordoba12** [comment]
Don't forget to change `/` to `\` before displaying filenames on Windows.

File: spyder_notebook/notebookplugin.py [hunk 1]
```diff
@@ -61,6 +63,10 @@ def __init__(self, parent):
 
         layout = QVBoxLayout()
 
+        filelist_btn = create_toolbutton(self, icon=ima.icon('filelist'),
+                                              tip=_("File list management"),
+                                              triggered=self.open_fileswitcher_dlg)
```
Comment (ccordoba12) [inline spyder_notebook/notebookplugin.py]: Please align the last two lines here to `(` in the first line

File: spyder_notebook/widgets/client.py [hunk 2]
```diff
@@ -115,7 +115,9 @@ class NotebookClient(QWidget):
     def __init__(self, plugin, name):
         super(NotebookClient, self).__init__(plugin)
 
-        self.name = name
+        if os.name == 'nt':
+            name = name.replace('/','\\')
+        self.filename = name
```
Comment (ccordoba12) [inline spyder_notebook/widgets/client.py]: Please leave this as it was, i.e. `self.name =  name`

Comment (ccordoba12) [inline spyder_notebook/widgets/client.py]: Sorry, it's fine as you wrote it :-)

Comment (ccordoba12) [inline spyder_notebook/widgets/client.py]: But to be consistent, you should change `name` to `filename` in `create_new_client`

**Comment by ccordoba12** [comment]
This still needs a test before we merge it :-)

**Comment by ccordoba12** [comment]
@dalthviz, we need to release Spyder **3.1.4** for your test to work. I'll try to do that tomorrow :-)

Besides, you need to change `setup.py` to demand `spyder>=3.1.4` so that this plugin has the right dependency.

**Comment by ccordoba12** [comment]
@dalthviz, please merge with master to fix the errors in Circle.

**Pull request merged**